client_app/src/components/admin/AddShelterForm.js

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from 'react';
+import '../../styles/admin/AddShelterForm.css';
+
+const AddShelterForm = () => {
+  const [formData, setFormData] = useState({
+    name: '',
+    street_address_1: '',
+    street_address_2: '',
+    city: '',
+    state: '',
+    postal_code: '',
+    latitude: '',
+    longitude: ''
+  });
+  
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [submitMessage, setSubmitMessage] = useState({ type: '', text: '' });
+  
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({
+      ...formData,
+      [name]: value
+    });
+  };
+  
+  const validateForm = () => {
+    const requiredFields = ['name', 'street_address_1', 'city', 'state', 'postal_code'];
+    for (const field of requiredFields) {
+      if (!formData[field]) {
+        setSubmitMessage({
+          type: 'error',
+          text: `${field.replace('_', ' ')} is required`
+        });
+        return false;
+      }
+    }
+    
+    //if given long and lat, checking this here otherwise give fomratting error
+    if (formData.latitude && (isNaN(formData.latitude) || formData.latitude < -90 || formData.latitude > 90)) {
+      setSubmitMessage({
+        type: 'error',
+        text: 'Latitude must be a number between -90 and 90'
+      });
+      return false;
+    }
+    
+    if (formData.longitude && (isNaN(formData.longitude) || formData.longitude < -180 || formData.longitude > 180)) {
+      setSubmitMessage({
+        type: 'error',
+        text: 'Longitude must be a number between -180 and 180'
+      });
+      return false;
+    }
+    
+    return true;
+  };
+  
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    
+    if (!validateForm()) {
+      return;
+    }
+    
+    setIsSubmitting(true);
+    setSubmitMessage({ type: '', text: '' });
+    
+    try {
+      const response = await fetch('/api/shelter', {
```

Comment:
```suggestion
      const response = await fetch(`${SERVER}/shelter, {
```
You should use the SERVER config variable for requests, otherwise the request might not go where it needs to go.
At the top of the file, add:
```import { SERVER } from "../../config";```